### ISSUE TITLE
Warning if the array value is assigned by index

### ISSUE THREAD

**Issue by ppn2** [MeirionHughes/aurelia-template-lint#82]
As per aurelia/binding#64 assigning an array value by index does not notify subscribers. It would be nice if a warning could be generated if this encountered.

**Comment by atsu85** [comment]
@ppn2, did You close this issue by accident?

**Comment by MeirionHughes** [comment]
urr nope, says you closed it. :P

**Comment by MeirionHughes** [comment]
What should we warn on?

👍 **viewModel**:  ```array[0] = 'bar'```
👎  **view**:  ```<div>${array[0]}</div>```
❤️  **both**  ```array[0] = 'bar' ```, only when a view binding to `array` is detected; 

cravet being the first two are relatively straight forward, but the third will need a separate rule I think.
